Enable E402 ruff check

peterfpeterson opened the pull request:
Pulls the following into `ornl-next`
* #38333
* #38350
* #38359
* #38314

robertapplin committed:
Enable ruff check E402 module level import

This is the changes from #38333

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/pre-commit/mirrors-clang-format: v19.1.2 → v19.1.3](https://github.com/pre-commit/mirrors-clang-format/compare/v19.1.2...v19.1.3)
- [github.com/astral-sh/ruff-pre-commit: v0.7.0 → v0.7.2](https://github.com/astral-sh/ruff-pre-commit/compare/v0.7.0...v0.7.2)

robertapplin committed:
Fix typo in mantid profiling link

robertapplin commented:
You might also need https://github.com/mantidproject/mantid/pull/38350 to fix the docs failure

peterfpeterson committed:
Do not allow qtpy v2.4.2

It changes the type of QMenu to be a sip object which fails an
isinstance() check a test. This is a quick fix to get the builds
working again. Follow on work to fix the test needs to happen.
